core/src/main/java/com/novoda/noplayer/player/PlayerFactory.java

```diff
@@ -58,10 +67,10 @@ public Player create(DrmType drmType, DrmHandler drmHandler, boolean downgradeSe
     private Player createPlayerForType(PlayerType playerType, DrmType drmType, DrmHandler drmHandler, boolean downgradeSecureDecoder) {
         switch (playerType) {
             case MEDIA_PLAYER:
-                return mediaPlayerCreator.createMediaPlayer(context);
+                return noPlayerMediaPlayerCreator.createMediaPlayer(context);
```

Comment:
Well, we can still have the `InternalCreator` it that benefits us for reducing the scope of those classes, but I still believe we can benefit from doing `player.initialise()` here instead of the two creators. 

Not a blocker, just a thought